cluster/query_executor.go

```diff
@@ -463,6 +466,12 @@ func (e *QueryExecutor) executeSelectStatement(stmt *influxql.SelectStatement, c
 	em.OmitTime = stmt.OmitTime
 	defer em.Close()
 
+	// Calculate initial stats across all iterators.
```

Comment:
The emitter is handling the closing of the iterators so I didn't want to do a double defer close. Really, the iterators should close themselves but that's a change for a different PR.